### PR TITLE
[matrix-js-sdk] add various types & clarify some old ones

### PR DESCRIPTION
**Pull request by holyslippers**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <http://matrix-org.github.io/matrix-js-sdk/9.2.0/index.html>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/matrix-js-sdk/index.d.ts [hunk 1]
```diff
@@ -270,7 +270,61 @@ export interface LoginPayload {
...
+  before_limit?: number;
+  after_limit?: number;
+  include_profile?: boolean;
+}
+export interface EventContext {
+  start: string;
+  end: string;
+  profile_info?: {
+    [userId: string]: {
+      displayname:	string;
```
Comment (danger-public) [inline types/matrix-js-sdk/index.d.ts:283]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by huan — APPROVED** [review]
LGTM

File: types/matrix-js-sdk/index.d.ts (hunk 1 above)
Comment (holyslippers) [reply types/matrix-js-sdk/index.d.ts:283]: Ok, I'll fix it

**Review by huan — APPROVED** [review]
LGTM

**Comment by holyslippers** [comment]
Thx @huan

**Comment by holyslippers** [comment]
Ready to merge

**Pull request merged**